### ISSUE TITLE
논문 리뷰

### ISSUE THREAD

**Issue by DongYuls** [DongYuls/markdown_playground#1]
## Exploring the Limits of Weakly Supervised Pre-training 

For more information check out the paper: [Mahajan, Dhruv, et al. "Exploring the Limits of Weakly Supervised Pretraining." *arXiv preprint arXiv:1805.00932* (2018).](https://arxiv.org/pdf/1805.00932.pdf) 

### Background

![](https://cdn-images-1.medium.com/max/799/1*tV_6-fK-G3wDs2DrybRgpw.jpeg)

- ImageNet is now nearly ten years old and is by modern standards “small”. **Relatively little is known about the behavior of pretraining** with datasets that are multiple orders of magnitude larger.
- Such datasets are dreadfully **labor-intensive** and **difficult to collect and annotate**.

### Contributions

![](https://9to5mac.files.wordpress.com/2017/12/follow-hashtag.jpg?quality=82&strip=all&w=1600)

This paper presents:

- **Unexplored data regime**: billions of images “labeled” in the wild with hashtags.
- **Improvements on transfer learning perfomance**: hashtags are too noisy, but the model trained on billions exhibits excellent transfer learning.

### Dataset

| Dataset Name  |    Role    |  # of Images  | # of Labels | Label Source  |
| :-----------: | :--------: | :-----------: | :---------: | :-----------: |
|   Instagram   |  Training  |  940,000,000  |    1,500    |   ImageNet    |
|   Instagram   |  Training  | 1,000,000,000 |    8,500    |    WordNet    |
|   Instagram   |  Training  | 1,000,000,000 |   17,000    |    WordNet    |
|   Instagram   |  Training  | 3,500,000,000 |   17,000    |    WordNet    |
|   ImageNet    |  Training  |   1,280,000   |    1,000    |   ImageNet    |
|   ImageNet    | Validation |    50,000     |    1,000    |   ImageNet    |
| Caltech Birds |  Training  |     6,000     |     200     | Caltech Birds |
| Caltech Birds | Validation |     6,000     |     200     | Caltech Birds |
|   Places365   |  Training  |   1,800,000   |     365     |   Places365   |
|   Places365   | Validation |    37,000     |     365     |   Places365   |
|     COCO      |  Training  |   1,350,000   |     80      |     COCO      |
|     COCO      | Validation |     5,000     |     80      |     COCO      |

- **Image deduplication**: R-MAC features for all candidate images using a ResNet-50 model, then $k=21$ nereast neighbors.
- Conservative lower bound on accuracy by marking all duplicates as incorrect.

### Architecture

Exsiting residual networks with grouped convolutional layers, ResNeXt, is purposely modified.

![](https://raw.githubusercontent.com/joshua19881228/my_blogs/master/Computer_Vision/Reading_Note/figures/ResNeXt.jpg)

- A block of ResNeXt with cardinality = 32, with roughly the same complexity. The layer shown above stands for (# in channels, filter size, # out channels).
- This paper uses **ResNeXt-101 32$\times C$d**, which has 101 layers, 32 groups (cardinalities), and groups widths $C$ of: {4, 8, 16, 32, 48}

![3](https://user-images.githubusercontent.com/29971008/42462045-17eb488c-83dd-11e8-9c83-adb97d9373e7.JPG)

- **Loss function**: Probabilities over all hashtags in the vocabulary using a softmax activation; minimizing the cross-entropy between the predicted softmax distribution and the target distribution of each image.
- Per-hashtag sigmoid outputs and binary logistic loss gives significantly worse results. (Why?)

### Experiments

Pretraining for hashtag prediction substantially improves target task accuracy.

![default](https://user-images.githubusercontent.com/29971008/42461637-f1432bba-83db-11e8-80d7-07a59d4d9af7.JPG)

**High bias!** Current network architectures are under-fitting when trained on billions of images. 

![1](https://user-images.githubusercontent.com/29971008/42461983-e7516fda-83dc-11e8-909b-8e5f5ec8ed39.png)

- **Generalization?** Increasing the size of the pre-training dataset is worthwhile, it is necessary to select a label space for the source task to match that of the target task. 

![2](https://user-images.githubusercontent.com/29971008/42461986-e8df2284-83dc-11e8-8a12-fef58dcda8fa.jpg)

### Conclusion

- Instagram has two nice properties: public visibility and simplicity.
- Compared to the traditional approach of manually designing the datasets, it works well!

---

## Learning to Diagnose from Scratch by Exploiting Dependencies Among Labels

For more information check out the paper: [Yao, Li, et al. "Learning to diagnose from scratch by exploiting dependencies among labels." *arXiv preprint arXiv:1710.10501* (2017).](https://arxiv.org/pdf/1710.10501.pdf ) 

### Abstract

- Many tasks in radiology, for example, are largely problems of multi-label classification wherein medical images are interpreted to indicate multiple present or suspected pathologies.
- Training model to recognize these interdependencies could enable better prediction of pathologic outcomes across all categories while maximizing the data utilization.

#### Independent prediction of labels

Ignoring the nature of conditional dependencies among the indicator, $$y^{t}$$, one could establish the following model:
$$
P(y|x) = \Pi_{t=1}^{T}P(y^{t}|x)
$$
where $y$ is a binary vector of dimensionality $T$, which is the total number of abnormalities, thus $y^{t}=0$ indicates its absence and $y^{t}=1$ its presence. 

#### Exploiting higher-order dependencies among labels

The following factorization, which makes no assumption of independence may suggest the absence or presence of others:
$$
P(y|x) = P(y^{0}|x)P(y^{1}|y^{0},x)P(y^{2}|y^{0},y^{1},x) ...P(y^{T}|y^{0},y^{1}, ..., y^{T-1}, x)
$$

However, the inference of such a model is unfortunately intractable as $y^∗=arg \min_{y} P(y^0,…,y^T |x)$. In practice, greedy search, which is equivalent to beam search with size 1, results in similar performance due to the binary sampling nature of each $y^t$. 

#### Model achitecture

Existing RNN are modified (paramerized as suggesting the absence or presence of others using factorization) to leverage interdependencies among in different pathologic patterns. 

![4](https://user-images.githubusercontent.com/29971008/42462684-eb958340-83de-11e8-8ecd-e08982dbd398.png)

- Standard sequence-to-sequence models often use softmax to predict one out of T classes and thus need to learn explicitly an “end-of-sequence” class.
- This is not desirable in our context due to the sparseness of the labels, resulting in the learned decoder being strongly biased towards predicting end-of-sequence while missing infrequently appearing abnormalities.

### Experiment

![5](https://user-images.githubusercontent.com/29971008/42462692-efefd74c-83de-11e8-816e-6ffa998c1ac8.png)

### Conclusion

- Complex interaction between abnormal pattern frequently have significant clinical meaning that provides radiologists with additional context.
- Exploiting dependencies among labels overcomes the scarcity of training data and maximizes the need to extract clinically relevant features from available samples – ideally without the pre-trained models which may carry undesirable biases.
